src/components/ReanimatedDrawerLayout.tsx

```diff
@@ -0,0 +1,750 @@
+    // 1) +---------------+ 2) +---------------+ 3) +---------------+ 4) +---------------+
+    //    |XXXXXXXX|......|    |XXXXXXXX|......|    |XXXXXXXX|......|    |XXXXX|.........|
+    //    |XXXXXXXX|......|    |XXXXXXXX|......|    |XXXXXXXX|......|    |XXXXX|.........|
+    //    |XXXXXXXX|..<*..|    |XXXXXXXX|.<-*..|    |XXXXXXXX|<--*..|    |XXXXX|<-----*..|
+    //    |XXXXXXXX|......|    |XXXXXXXX|......|    |XXXXXXXX|......|    |XXXXX|.........|
+    //    |XXXXXXXX|......|    |XXXXXXXX|......|    |XXXXXXXX|......|    |XXXXX|.........|
+    //    +---------------+    +---------------+    +---------------+    +---------------+
+
+    const openValue = useDerivedValue(() => {
+      const newOpenValue = interpolate(
+        drawerTranslation.value,
+        [0, drawerWidth],
+        [0, 1],
+        Extrapolation.CLAMP
+      );
+
+      props.onDrawerSlide && runOnJS(props.onDrawerSlide)(newOpenValue);
+
+      return newOpenValue;
+    });
+
+    const isDrawerOpen = useSharedValue(false);
+
+    const handleContainerLayout = ({ nativeEvent }: LayoutChangeEvent) => {
+      setContainerWidth(nativeEvent.layout.width);
+    };
+
+    const emitStateChanged = useCallback(
+      (newState: DrawerState, drawerWillShow: boolean) => {
+        'worklet';
+        props.onDrawerStateChanged &&
+          runOnJS(props.onDrawerStateChanged)?.(newState, drawerWillShow);
+      },
+      [props.onDrawerStateChanged]
+    );
+
+    const drawerAnimatedProps = useAnimatedProps(() => ({
+      accessibilityViewIsModal: isDrawerOpen.value,
+    }));
+
+    const overlayAnimatedProps = useAnimatedProps(() => ({
+      pointerEvents: isDrawerOpen.value ? ('auto' as const) : ('none' as const),
+    }));
+
+    // While the drawer is hidden, it's hitSlop overflows onto the main view by edgeWidth
+    // This way it can be swiped open even when it's hidden
+    const [edgeHitSlop, setEdgeHitSlop] = useState<HitSlop>(
+      isFromLeft
+        ? { left: 0, width: edgeWidth }
+        : { right: 0, width: edgeWidth }
+    );
+
+    // gestureOrientation is 1 if the gesture is expected to move from left to right and -1 otherwise
+    const gestureOrientation = useMemo(
+      () => sideCorrection * (drawerOpened ? -1 : 1),
+      [sideCorrection, drawerOpened]
+    );
+
+    const animateDrawer = useCallback(
+      (toValue: number, initialVelocity: number, animationSpeed?: number) => {
+        'worklet';
+        const willShow = toValue !== 0;
+        isDrawerOpen.value = willShow;
+        runOnJS(setEdgeHitSlop)(
+          isFromLeft
+            ? { left: 0, width: willShow ? undefined : edgeWidth }
+            : { right: 0, width: willShow ? undefined : edgeWidth }
+        );
+
+        emitStateChanged(DrawerState.SETTLING, willShow);
+        runOnJS(setDrawerState)(DrawerState.SETTLING);
+
+        if (props.hideStatusBar) {
+          StatusBar.setHidden(willShow, statusBarAnimation);
```

Comment:
Does this work on the UI thread? I don't think `setHidden` is a worklet.

Reply:
Fixed in [900fa4f](https://github.com/software-mansion/react-native-gesture-handler/pull/3146/commits/900fa4f147164805097eb0da6ad83ce2b8c30ca7)